libcloud/dns/drivers/pointdns.py

```diff
@@ -332,9 +371,17 @@ def update_record(self, record, name, type, data, extra=None):
         if extra is not None:
             r_json.update(extra)
         r_data = json.dumps({'zone_record': r_json})
-        response = self.connection.request('/zones/%s/records/%s' %
-                                           (zone.id, record.id),
-                                           method='PUT', data=r_data)
+        try:
+            response = self.connection.request('/zones/%s/records/%s' %
+                                               (zone.id, record.id),
+                                               method='PUT', data=r_data)
+        except (BaseHTTPError, MalformedResponseError):
+            e = sys.exc_info()[1]
+            if isinstance(e, MalformedResponseError) and e.body == 'Not found':
+                raise RecordDoesNotExistError(value="Record doesn't exists",
```

Comment:
Another "for the future reference" - Ideally those exceptions would be handled on the Response class - see how other classes use `set_context` method.